### PR TITLE
hide boost::bimap using pimpl and proxy design pattern

### PR DESCRIPTION
**Pull request by prudhomm**
this is done in submeshdata and gmshordering
see #1473

- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] _Have you added an explanation of what your changes do and why you'd like us to include them?_
yes in #1473 

- [ ] _Have you written new tests for your changes?_
No

- [X] _Have you successfully run the Feel++ testsuite with your changes locally?_
Only the quickstart testsuite was run

- [ ] Have you written Doxygen comments in your contribution ?

-----
See #1473

### CODE REVIEW

**Comment by prudhomm** [comment]
@vincentchabannes could you review ?

File: feelpp/feel/feelcore/bimap.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+bimap<LeftT, RightT>::leftFind( left_t i ) const
+{
+    if ( auto l = pimpl_->bm.left.find( i ); l != pimpl_->bm.left.end() )
+        return l->second;
+    else  
+        return std::nullopt;
+}
+template<typename LeftT, typename RightT>
+std::optional<typename bimap<LeftT, RightT>::left_t>
+bimap<LeftT, RightT>::rightFind( right_t i ) const
```
Comment (vincentchabannes) [inline feelpp/feel/feelcore/bimap.cpp:87]: idem

File: feelpp/feel/feelcore/bimap.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+{
+    if ( auto l = pimpl_->bm.right.find( i ); l != pimpl_->bm.right.end() )
+        return l->second;
+    else  
+        return std::nullopt;
+}
+template<typename LeftT, typename RightT>
+typename bimap<LeftT, RightT>::right_t
+bimap<LeftT, RightT>::leftAt( left_t i ) const
```
Comment (vincentchabannes) [inline feelpp/feel/feelcore/bimap.cpp:96]: idem

**Review by vincentchabannes — COMMENTED** [review]
@prudhomm There are also some inclusions of bimap in `feel/feelfilters/hbf.hpp` (this one was problematic on tgcc if I remember) and in feelcrb

**Pull request closed without merging**